Aplib.Core/IHasMetadata.cs

```diff
@@ -0,0 +1,16 @@
+namespace Aplib.Core
+{
+    /// <summary>
+    /// Represents an object that contains an instance of <see cref="IMetadata"/>.
+    /// </summary>
+    public interface IHasMetadata
```

Comment:
`IHasMetadata` is an unconventional way of creating interfaces. If we look at the C# standard library, we never see an interface contain 'has'. I would like to propose another name for the interface: `IDocumented`. This name clearly communicates the requirements for an object to have metadata, and conforms to the format many libraries use.

Reply:
Agreed